cwl/src/main/scala/cwl/CwlExpressionCommandPart.scala

```diff
@@ -43,3 +44,29 @@ case class CwlArgumentCommandPart(argument: CommandLineBinding) extends CommandP
   }
 }
 
+case class InputParameterCommandPart(argument: CommandInputParameter) extends CommandPart {
+
+  override def instantiate(inputsMap: Map[LocalName, WomValue],
+                           functions: IoFunctionSet,
+                           valueMapper: (WomValue) => WomValue) = {
+
+    val localInputsMap = inputsMap.map{
```

Comment:
ToL: Rather than preprocessing the entire list ahead of time, why not make a method (`eg "formatForCommandInsertion(womValue: WomValue)"`) and call it once we know which value it is we want to insert (on line 62)?